File: app/utils/logging.py
from __future__ import annotations

import logging
import sys

from ..config import get_settings

settings = get_settings()


def setup_logging() -> None:
    log_level = logging.DEBUG if settings.DEBUG else logging.INFO
    logging.basicConfig(
        level=log_level,
        format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
        handlers=[logging.StreamHandler(sys.stdout)],
    )

    logging.getLogger("uvicorn").setLevel(logging.INFO)
    logging.getLogger("aiosmtpd").setLevel(logging.INFO)
    logging.getLogger("redis").setLevel(logging.WARNING)


def get_logger(name: str) -> logging.Logger:
    return logging.getLogger(f"mapae.{name}")

File: internal/logging/logging.go
package logging

import (
	"log"
	"os"
)

type Logger struct {
	*log.Logger
}

func New(prefix string, debug bool) *Logger {
	flags := log.LstdFlags
	if debug {
		flags = log.LstdFlags | log.Lshortfile
	}
	return &Logger{Logger: log.New(os.Stdout, prefix, flags)}
}
